assets/js/modules/analytics/datastore/properties.test.js

```diff
@@ -123,6 +123,67 @@ describe( 'modules/analytics properties', () => {
 				expect( properties ).toEqual( undefined );
 			} );
 		} );
+		describe( 'selectProperty', () => {
+			it( 'requires a valid propertyID', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).selectProperty();
+				} ).toThrow( 'A valid propertyID selection is required.' );
+			} );
+
+			it( 'returns if the accountID is not set', () => {
+				fetchMock.get(
+					/^\/google-site-kit\/v1\/modules\/analytics\/data\/properties-profiles/,
+					{ body: fixtures.propertiesProfiles, status: 200 }
+				);
+				const propertyID = fixtures.propertiesProfiles.properties[ 0 ].id;
+				expect( registry.select( STORE_NAME ).getPropertyID() ).toBeUndefined();
+				registry.dispatch( STORE_NAME ).selectProperty( propertyID );
+				expect( registry.select( STORE_NAME ).getPropertyID() ).toBeUndefined();
+			} );
+
+			it( 'selects the property', async () => {
+				fetchMock.get(
+					/^\/google-site-kit\/v1\/modules\/analytics\/data\/properties-profiles/,
+					{ body: fixtures.propertiesProfiles, status: 200 }
+				);
+				const accountID = fixtures.propertiesProfiles.properties[ 0 ].accountId;
+				const propertyID = fixtures.propertiesProfiles.properties[ 0 ].id;
+				registry.dispatch( STORE_NAME ).setAccountID( accountID );
+				registry.dispatch( STORE_NAME ).selectProperty( propertyID );
+
+				await subscribeUntil( registry,
+					() => (
+						registry.select( STORE_NAME ).getProperties( accountID ) !== undefined
+					),
+				);
+				expect( registry.select( STORE_NAME ).getPropertyID() ).toMatch( propertyID );
+				expect( registry.select( STORE_NAME ).getProperties( accountID ) ).toEqual( fixtures.propertiesProfiles.properties );
+				expect( registry.select( STORE_NAME ).getInternalWebPropertyID() ).toEqual( fixtures.propertiesProfiles.properties[ 0 ].internalWebPropertyId );
+				expect( registry.select( STORE_NAME ).getProfileID() ).toEqual( fixtures.propertiesProfiles.properties[ 0 ].defaultProfileId );
+			} );
+
+			it( 'does not set the propertyID if default is not found', async () => {
```

Comment:
Do you mean the profileID? It's a bit unclear here as you're asserting that the `propertyID` is empty which would mean the property was not found in state.

We should also have a test for making sure that if `selectProperty` is called with a property ID of a property that _has_ a `defaultProfileId` set but no matching profile in state (because it may not exist), then the `profileID` should _not be set_ to the property's `defaultProfileId`. That may be what you're aiming for here but just want to make sure we have this covered as well.